test_financials.py

```diff
@@ -0,0 +1,24 @@
+import unittest
+from unittest.mock import Mock
+from unittest.mock import patch
+import yfinance as yf
+
+class FinancialsTestCase(unittest.TestCase):
+    def setUp(self):
+        self.msft = yf.Ticker("MSFT")
+        self.utils_mock = patch('yfinance.base.utils.get_json', autospec=True).start()
+        self.addCleanup(patch.stopall)
+
+    def tearDown(self):
+        pass
+
+    def test_get_financials_invalid_data_returns_empty(self):
+        try:
+            financials = self.msft.financials 
+            self.assertTrue(financials.empty)
+        except:
```

Comment:
I'm not quite sure if I understand why we need an except statement here.

Couldn't we just use lines 17-18 without a try-except? I ran your test without it and it seems to run fine (no failures)

Reply:
oops just saw that this was merged lol

Reply:
There should be a comment on line 20, but the reason for the try/except is so to ensure an exception is NOT raised because we want graceful behavior of the application. We should not be throwing an exception even with invalid data, but simply returning an empty property for financials.

Reply:
Hmmm I see, so it's more so to ensure the test runs smoothly then. I think that's more clear now, thanks 👍